lib/util/http/github.ts

```diff
@@ -0,0 +1,302 @@
+}
+
+export interface GithubHttpOptions extends InternalHttpOptions {
+  paginate?: boolean | string;
+  pageLimit?: number;
+  token?: string;
+}
+
+interface GithubGraphqlResponse<T = unknown> {
+  data?: {
+    repository?: T;
+  };
+  errors?: { message: string; locations: unknown }[];
+}
+
+function handleGotError(
+  err: GotRequestError,
+  url: string | URL,
+  opts: GithubHttpOptions
+): never {
+  const path = url.toString();
+  let message = err.message || '';
+  if (err.body?.message) {
+    message = err.body.message;
+  }
+  if (
+    err.name === 'RequestError' &&
+    (err.code === 'ENOTFOUND' ||
+      err.code === 'ETIMEDOUT' ||
+      err.code === 'EAI_AGAIN')
+  ) {
+    logger.debug({ err }, 'GitHub failure: RequestError');
+    throw new Error(PLATFORM_FAILURE);
+  }
+  if (err.name === 'ParseError') {
+    logger.debug({ err }, 'GitHub failure: ParseError');
+    throw new Error(PLATFORM_FAILURE);
+  }
+  if (err.statusCode >= 500 && err.statusCode < 600) {
+    logger.debug({ err }, 'GitHub failure: 5xx');
+    throw new Error(PLATFORM_FAILURE);
+  }
+  if (
+    err.statusCode === 403 &&
+    message.startsWith('You have triggered an abuse detection mechanism')
+  ) {
+    logger.debug({ err }, 'GitHub failure: abuse detection');
+    throw new Error(PLATFORM_RATE_LIMIT_EXCEEDED);
+  }
+  if (err.statusCode === 403 && message.includes('Upgrade to GitHub Pro')) {
+    logger.debug({ path }, 'Endpoint needs paid GitHub plan');
+    throw err;
+  }
+  if (err.statusCode === 403 && message.includes('rate limit exceeded')) {
+    logger.debug({ err }, 'GitHub failure: rate limit');
+    throw new Error(PLATFORM_RATE_LIMIT_EXCEEDED);
+  }
+  if (
+    err.statusCode === 403 &&
+    message.startsWith('Resource not accessible by integration')
+  ) {
+    logger.debug(
+      { err },
+      'GitHub failure: Resource not accessible by integration'
+    );
+    throw new Error(PLATFORM_INTEGRATION_UNAUTHORIZED);
+  }
+  if (err.statusCode === 401 && message.includes('Bad credentials')) {
+    const rateLimit = err.headers ? err.headers['x-ratelimit-limit'] : -1;
+    logger.debug(
+      {
+        token: maskToken(opts.token),
+        err,
+      },
+      'GitHub failure: Bad credentials'
+    );
+    if (rateLimit === '60') {
+      throw new Error(PLATFORM_FAILURE);
+    }
+    throw new Error(PLATFORM_BAD_CREDENTIALS);
+  }
+  if (err.statusCode === 422) {
+    if (
+      message.includes('Review cannot be requested from pull request author')
+    ) {
+      throw err;
+    } else if (
+      err.body &&
+      err.body.errors &&
+      err.body.errors.find((e: any) => e.code === 'invalid')
+    ) {
+      throw new Error(REPOSITORY_CHANGED);
+    }
+    logger.debug({ err }, '422 Error thrown from GitHub');
+    throw new Error(PLATFORM_FAILURE);
+  }
+  throw err;
+}
+
+export class GithubHttp extends Http<GithubHttpOptions, GithubHttpOptions> {
+  constructor(
+    hostType: string = PLATFORM_TYPE_GITHUB,
```

Comment:
```suggestion
```

Let's maybe hardcode `hostType` instead of making it configurable. Any downsides?